api/src/main/java/org/openmrs/module/addresshierarchy/config/AddressConfigurationLoader.java

```diff
@@ -0,0 +1,274 @@
+			AddressHierarchyLevel lastLevel = null;
+			for (AddressComponent component : addressComponents) {
+				AddressHierarchyLevel level = new AddressHierarchyLevel();
+				level.setName(component.getNameMapping());
+				level.setAddressField(component.getField());
+				level.setRequired(component.isRequiredInHierarchy());
+				level.setParent(lastLevel);
+				getService().saveAddressHierarchyLevel(level);
+				lastLevel = level;
+			}
+		}
+		else {
+
+			log.info("Updating address hierarchy levels");
+
+			for (AddressComponent component : addressComponents) {
+				AddressHierarchyLevel level = getService().getAddressHierarchyLevelByAddressField(component.getField());
+				level.setRequired(component.isRequiredInHierarchy());
+				getService().saveAddressHierarchyLevel(level);
+			}
+		}
+	}
+
+	/**
+	 * Install the address hierarchy entries as defined by the AddressHierarchyFile configuration
+	 */
+	public static void installAddressHierarchyEntries(AddressHierarchyFile file, boolean deleteEntries) {
+		log.info("Installing address hierarchy entries");
+		if (deleteEntries) {
+			log.warn("Deleting existing address hierarchy entries");
+			getService().deleteAllAddressHierarchyEntries();
+		}
+		
+		InputStream is = null;
+		try {
+			is = new FileInputStream( ConfigLoaderUtil.getFile(getSubdirConfigPath(), file.getFilename()) );
+			AddressHierarchyImportUtil.importAddressHierarchyFile(is, file.getEntryDelimiter(), file.getIdentifierDelimiter());
+		}
+		catch (Exception e) {
+			throw new IllegalArgumentException("Unable to import address hierarchy from file", e);
+		}
+		finally {
+			IOUtils.closeQuietly(is);
+		}
+	}
+
+	/**
+	 * Reads from a String representing the address configuration into an AddressConfiguration object
+	 */
+	public static AddressConfiguration readFromFile(File file) {
+		try {
+			String configuration = FileUtils.readFileToString(file, "UTF-8");
+			return readFromString(configuration);
+		}
+		catch (IOException e) {
+			throw new IllegalArgumentException("Unable to load address configuration from configuration file.  Please check the format of this file", e);
+		}
+	}
+
+	/**
+	 * Reads from a String representing the address configuration into an AddressConfiguration object
+	 */
+	public static AddressConfiguration readFromString(String configuration) {
+		try {
+			return (AddressConfiguration) getSerializer().fromXML(configuration);
+		}
+		catch (Exception e) {
+			throw new IllegalArgumentException("Unable to load address configuration from configuration file.  Please check the format of this file", e);
+		}
+	}
+
+	/**
+	 * Writes a serialized String representing the address configuration from an AddressConfiguration object
+	 */
+	public static String writeToString(AddressConfiguration configuration) {
+		return getSerializer().toXML(configuration);
+	}
+
+	/**
+	 * @return the serializer instance used to load configuration from file
+	 */
+	public static XStream getSerializer() {
```

Comment:
Since we have our own custom serializer instance here, we should be able to have a module-defined AddressTemplate class, that serializes in a compatible way with the core serializer, and not require reflection.